lib/assocPath/assocPath.test.js

```diff
@@ -0,0 +1,20 @@
+var assocPath = require('.')
+
+test('creates nesting object associated with given path and places given value into its tail', () => {
+  const obj = { a: { b: { c: 0 }}}
+  expect(assocPath(['a', 'b', 'c'], 42, obj)).toEqual({ a: { b: { c: 42 }}})
+  expect(assocPath(['a', 'b', 'c'])(42, obj)).toEqual({ a: { b: { c: 42 }}})
+  expect(assocPath(['a', 'b', 'c'])(42)(obj)).toEqual({ a: { b: { c: 42 }}})
+  expect(assocPath(['a', 'b', 'c'], 42)(obj)).toEqual({ a: { b: { c: 42 }}})
+})
+
+test('overrides any value defined in object, if it\'s key provided in path', () => {
+  const obj = { a: 1 }
+  expect(assocPath(['a', 'b', 'c'], 42, obj)).toEqual({ a: { b: { c: 42 }}})
+})
+
+test('makes shallow copy of object, not copying by ref', () => {
+  const obj = { a: { b: 0 }}
+  const newObj = assocPath(['a', 'b'], 42, obj)
+  expect(obj === newObj).toBeFalsy()
```

Comment:
I would also add check for `expect(obj.a == newObj.a).toBeTruthy()`

Reply:
agree

Reply:
hm, although, I think we should only check for non-edited data, like:

```js
const obj = { a: { b: 0 }, c: {} }
const newObj = assocPath(['a', 'b'], 42, obj)

expect(obj.c === newObj.c).toBeThuthy() //check c, a has been edited - not mutated, but cloned, so
```

Checking `c` prop of object, because `a` has been edited - we'v made new copy, not mutated it

@Beraliv  Your opinion?

Reply:
As `ramda` has full copy and it's logical to do so, I would suggest to check for inequality by reference